src/dom.js

```diff
@@ -13,35 +13,228 @@
+ */
+const ElementExtensions = {
+  /** @extends htmlparser2.Element.prototype */
+
+  nodeName: {
+    get() {
+      return this.tagName.toUpperCase();
+    },
+  },
+
+  id: reflectedProperty('id'),
+
+  className: reflectedProperty('class'),
+
+  insertBefore(child, referenceNode) {
+    if (!referenceNode) return this.appendChild(child);
+    treeAdapter.insertBefore(this, child, referenceNode);
+    return child;
+  },
+
+  appendChild(child) {
+    treeAdapter.appendChild(this, child);
+    return child;
+  },
+
+  removeChild(child) {
+    treeAdapter.detachNode(child);
+  },
+
+  remove() {
+    treeAdapter.detachNode(this);
+  },
+
+  textContent: {
+    get() {
+      return this.children.map((node) => node.nodeValue).join('\n');
+    },
+
+    set(text) {
+      this.children = [];
+      treeAdapter.insertText(this, text);
+    },
+  },
+
+  setAttribute(name, value) {
+    if (this.attribs == null) this.attribs = {};
+    if (value == null) value = '';
+    this.attribs[name] = value;
+  },
+
+  removeAttribute(name) {
+    if (this.attribs != null) {
+      delete this.attribs[name];
+    }
+  },
+
+  getAttribute(name) {
+    return this.attribs != null && this.attribs[name];
+  },
+
+  hasAttribute(name) {
+    return this.attribs != null && this.attribs[name] != null;
+  },
+
+  getAttributeNode(name) {
+    const value = this.getAttribute(name);
+    if (value != null) return { specified: true, value };
+  },
+};
+
+/**
+ * Methods and descriptors to mix into the global document instance
+ * @private
+ */
+const DocumentExtensions = {
+  /** @extends htmlparser2.Document.prototype */
+
+  // document is just an Element in htmlparser2, giving it a nodeType of ELEMENT_NODE.
+  // TODO: verify if these are needed for css-select
+  nodeType: {
+    get() {
+      return 9;
+    },
+  },
+
+  contentType: {
+    get() {
+      return 'text/html';
+    },
+  },
+
+  nodeName: {
+    get() {
+      return '#document';
+    },
+  },
+
+  documentElement: {
+    get() {
+      // Find the first <html> element within the document
+      return this.childNodes.filter(
+        (child) => String(child.tagName).toLowerCase() === 'html'
+      );
+    },
+  },
+
+  compatMode: {
+    get() {
+      const compatMode = {
+        'no-quirks': 'CSS1Compat',
+        quirks: 'BackCompat',
+        'limited-quirks': 'CSS1Compat',
+      };
+      return compatMode[treeAdapter.getDocumentMode(this)];
+    },
+  },
+
+  body: {
+    get() {
+      return this.querySelector('body');
+    },
+  },
+
+  createElement(name) {
+    return treeAdapter.createElement(name, null, []);
+  },
+
+  createTextNode(text) {
+    // there is no dedicated createTextNode equivalent exposed in htmlparser2's DOM
+    const Node = this.$$Node;
+    return new Node({
+      type: 'text',
+      data: text,
+      parent: null,
+      prev: null,
+      next: null,
+    });
+  },
+
+  querySelector(sel) {
+    if (sel === ':root') {
+      return this.type === 'root';
```

Comment:
Seems like this is returning a Boolean when it should be returning the nearest document root.

```suggestion
      return this;
```